client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -173,72 +150,59 @@ module.exports = React.createClass( {
 							count: this.getNumberOfDomainRegistrations(),
 							textOnly: true
 						}
-					)
-				} ) }
-				{ this.field( Input, {
-					name: 'email',
-					label: this.translate( 'Email', { textOnly: true } )
-				} ) }
-				{ this.field( Input, {
-					name: 'phone',
-					label: this.translate( 'Phone', { textOnly: true } )
-				} ) }
-				{ this.field( CountrySelect, {
-					name: 'country-code',
-					label: this.translate( 'Country', { textOnly: true } ),
-					countriesList: countriesList
-				} ) }
-				{ this.showFax() ? this.field( Input, {
-					name: 'fax',
-					label: this.translate( 'Fax', { textOnly: true } )
-				} ) : null }
-				{ this.field( Input, {
-					name: 'address-1',
-					label: this.translate( 'Address', { textOnly: true } )
-				} ) }
-				{ this.field( HiddenInput, {
-					name: 'address-2',
-					label: this.translate( 'Address Line 2', { textOnly: true } ),
-					text: this.translate( '+ Add Address Line 2', { textOnly: true } )
-				} ) }
-				{ this.field( Input, {
-					name: 'city',
-					label: this.translate( 'City', { textOnly: true } )
-				} ) }
-				{ this.field( StateSelect, {
-					name: 'state',
-					label: this.translate( 'State', { textOnly: true } ),
-					countryCode: countryCode,
-					statesList: statesList
-				} ) }
-				{ this.field( Input, {
-					name: 'postal-code',
-					label: this.translate( 'Postal Code', { textOnly: true } )
-				} ) }
+					) }
+					{ ...fieldProps( 'organization' ) }/>
+
+				<Input label={ this.translate( 'Email', { textOnly } ) } { ...fieldProps( 'email' ) }/>
+				<Input label={ this.translate( 'Phone', { textOnly } ) } { ...fieldProps( 'phone' ) }/>
+
+				<CountrySelect
+					label={ this.translate( 'County', { textOnly } ) }
+					countriesList={ countriesList }
+					{ ...fieldProps( 'country-code' ) }/>
+
+				{ this.showFax() && <Input label={ this.translate( 'Fax', { textOnly } ) } { ...fieldProps( 'fax' ) }/> }
+				<Input label={ this.translate( 'Address', { textOnly } ) } { ...fieldProps( 'address-1' ) }/>
+
+				<Input
```

Comment:
This field should be a `HiddenInput`, as it is in the original code, so that the `+ Add Address Line 2` link is actually shown and works as intended.